File: evidence/generated-systems/angular/governance-on-angular/src/app/models/AuditWorkpaper.js

// Define collection and schema for AuditWorkpaper
export interface AuditWorkpaper {
    workpaperRef:
	type : string
    subject:
	type : string
    workpaperUrl:
	type : URL
    Engagement:
	type : Schema.Types.ObjectId
    Evidence:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Evidence' }]
    Findings:
 	type : [{ type: Schema.Types.ObjectId, ref: 'AuditFinding' }]
#
    collection: 'auditWorkpapers'
}

File: evidence/generated-systems/rubyonrails/governance-on-rails/governanceonrails/app/models/auditworkpaper.rb
class AuditWorkpaper < ApplicationRecord


  composed_of :uRL,
    class_name: "URL",
    mapping: [
      %w[uRL_value value]
    ]

  has_many :Engagement, class_name: 'AuditEngagement'
  has_many :Evidence, class_name: 'Evidence'
  has_many :Findings, class_name: 'AuditFinding'

end
